core/src/main/scala/org/ensime/indexer/ClassfileIndexer.scala

```diff
@@ -179,11 +174,26 @@ final class ClassfileIndexer(file: FileObject) extends SLF4JLogging {
     ) = handleAnn(desc)
   }
 
+  private trait ReferenceInFieldHunter {
+    this: FieldVisitor =>
+
+    protected var internalRefs = Set.empty[FullyQualifiedName]
+
+    private val annVisitor = new AnnotationVisitor(ASM5) {
+      override def visitAnnotation(name: String, desc: String): AnnotationVisitor = handleAnn(desc)
+    }
+    private def handleAnn(desc: String): AnnotationVisitor = {
+      internalRefs = internalRefs + ClassName.fromDescriptor(desc)
+      annVisitor
+    }
+    override def visitTypeAnnotation(typeRef: Int, typePath: TypePath, desc: String, visible: Boolean): AnnotationVisitor = handleAnn(desc)
+    override def visitAnnotation(desc: String, visible: Boolean): AnnotationVisitor = handleAnn(desc)
+  }
+
   private trait ReferenceInMethodHunter {
     this: MethodVisitor =>
 
-    // NOTE: :+ and :+= are really slow (scala 2.10), prefer "enqueue"
-    protected var internalRefs = Queue.empty[FullyQualifiedName]
+    protected var internalRefs = Set.empty[FullyQualifiedName]
```

Comment:
same again. We'd profiled this and Queue was pretty fast... using scala Sets is surely going to slow things down